### ISSUE TITLE
containsExactly and containsExactlyInAnyOrder should also work for substrings

### ISSUE THREAD

**Issue by lludwa** [joel-costigliola/assertj-core#1063]
#### Summary
AssertJ has very convenient methods ```containsExactly``` and ```containsExactlyInAnyOrder``` for testing Collections.
On the other hand, it can also test substrings using ```containsSequence``` but is limited only to the exact sequence of Strings

I would like to have the ability to assert, that given substrings contains exactly expected result (like ```containsSequence```) but contains it exactly in any order.

Maybe instead of ```containsSequence``` you could overload ```containsExactly``` and ```containsExactlyInAnyOrder``` to work for substrings as well?

#### Example

```java
 assertThat("Element1 Element2").containsExactlyInAnyOrder("Element2", "Element1");
```
#### Java 8 specific ?
* NO

**Comment by joel-costigliola** [comment]
I'm not sure to get what you mean by exact, in your example `containsExactlyInAnyOrder` as I understand it would not work because the `"Element1 Element2"` has a space whereas `"Element2", "Element1"` don't have any.
For me the following assertions would work:
```java
assertThat("Element1 Element2").containsExactlyInAnyOrder("Element2", "Element1 ");
assertThat("Element1 Element2").containsExactlyInAnyOrder(" Element2", "Element1");
assertThat("Element1 Element2").containsExactlyInAnyOrder("Element2", "Element1", " ");
```

**Comment by lludwa** [comment]
Yes, you are right - that is exactly what I ment

**Comment by joel-costigliola** [comment]
Do you see a value for `containsExactly` ? I mean, if you know the substrings and their order you can simply use `isEqualTo`  and concatenate all the substrings

**Comment by lludwa** [comment]
Just for sake of consistency.
Also ```containsSequence``` will not be misleading anymore.

**Comment by joel-costigliola** [comment]
What is misleading with `containsSequence` ?

**Comment by lludwa** [comment]
because it rather matches "sub sequence" than "whole sequence"

**Comment by joel-costigliola** [comment]
true, I think we will rename this to `containsSubsequence` and add a proper `containsSequence`.

**Comment by eximius313** [comment]
this feature also looks like "[breakin change](https://github.com/joel-costigliola/assertj-core/issues?q=label%3A"breaking+change")"

**Comment by eximius313** [comment]
@joel-costigliola 
this feature also looks like "[breaking change](https://github.com/joel-costigliola/assertj-core/issues?q=label%3A"breaking+change")"

**Comment by joel-costigliola** [comment]
yes indeed

**Comment by eximius313** [comment]
So maybe it will be good to include it in 2.9.0 (as 2.9.0 already has some breaking changes)?

**Comment by joel-costigliola** [comment]
I want to release 2.9.0 by the end of the year, if I keep adding stuff I will never make it so unless someone is contributing this, it will be in the release after 2.9.0

**Comment by BillyYccc** [comment]
Hi! I'm working on [this](https://github.com/BillyYccc/assertj-core/tree/issue/1063). I'm a little confused by the intention of these two methods.
In my opinion
1.  `containsExactly()` means the ` actual charSequence` totally equals to the sequentially concat `charSequence...` with the comparison strategy.
this works  `assertThat("FOO BAR").containsExactly("FOO"," ","BAR")`

2. `containsExactlyInAnyOrder()` means `actual charSequence` totally contains all the elements(character) of the `charSequence...` (no more or less) with the comparison strategy.
this works `assertThat("foobar").containsExactlyInAnyOrder("r","a","b","o","o","f")`

Please correct me if i have a wrong understanding.

Besides, I don't find the breaking changes in this issue. The changes in `containsSequence` and `containsSubsequence ` have been fixed in [commit 5ff80a...](https://github.com/joel-costigliola/assertj-core/commit/5ff80a51c3a8284ffb8d52c9cc3b0bfd0b259dc1)

**Comment by eximius313** [comment]
I think it is well described in this comment: https://github.com/joel-costigliola/assertj-core/issues/1063#issuecomment-326043743
